### DISCUSSION TITLE
Navigation Buggy after Updating to 2.6.1

### DISCUSSION THREAD

**Discussion by bomshteyn** [livewire/livewire#3725] [7. Bugs (archived)] [answered]
The URL jumps back to previous page about a second after navigating between livewire pages while showing the correct content.

Downgrading to livewire 2.5.5 solves the issue.

**Comment by joshhanley** [comment]
@bomshteyn thanks for reporting! Can you please provide code so we can replicate the issue and any steps that you followed (such as buttons to click, etc)?

Thanks!

**Reply by bomshteyn** [reply]
The codebase is pretty complex with a lot of livewire components on every page. So i am not sure what i can provide in terms of code to be helpful.

**Reply by bomshteyn** [reply]
The navigation is using regular `a` tags and shows the correct page but the url in the nav bar just jumps back about a second later

**Reply by joshhanley** [reply]
@bomshteyn just anything so I can replicate the issue would be great! 🙂

**Reply by bomshteyn** [reply]
I narrowed it down to pages that are using [Livewire Tables Package](https://github.com/rappasoft/laravel-livewire-tables) 
so maybe it's somehow related to it.

**Reply by cswni** [reply]
The page related to link load correctly but the url path show the previous page visited. If you refresh the page then load the previous page.
Im using Livewire 2.6.6 and [Datatables Livewire Package](https://github.com/rappasoft/laravel-livewire-tables)

**Reply by Krato** [reply]
I tjink is related to https://github.com/livewire/livewire/discussions/3734. Maybe that issue can help you to check the error.

**Reply by joshhanley** [reply]
Yeah we fixed some issues with query strings on nested components, but maybe it messed something else up.

I'm currently working through @Krato's issue to see if I can replicate the problem.

If I can replicate it, then we can look at getting it fixed 🙂

**Comment by BostjanOb** [comment]
https://github.com/BostjanOb/LivewireBugDemo

Here is a sample project.

How to replicate:

1. Open the main page
2. Type "something" in input and submit it

The expected result should be: `/search/all?q=something`

Given result: `/?q=something&group=`

 #3524  changes to `js/comonent/SyncBrowserHistroy.js` introduced this bug

**Comment by joshhanley** [comment, marked as answer]
I have tagged v2.6.2 which fixes this issue 🙂

Please test it and let us know if it's still happening.

Special thanks to @BostjanOb for the demo repo!

Thanks!

**Reply by michele-grifa** [reply]
The issue is resolved for me. Thanks for your help.

**Reply by joshhanley** [reply]
No worries 🙂 thanks for confirming

**Reply by bomshteyn** [reply]
Thank you @BostjanOb  & @joshhanley !

**Reply by joshhanley** [reply]
@bomshteyn no worries 🙂

**Comment by rizkhal** [comment]
Hello, i think i have the similliar problem, but this problem only on production.

The url always reset to root url after i change, search or do anything about livewire request.. I have the demo video below.

Thanks

https://user-images.githubusercontent.com/24653114/135790757-b75c2141-f88f-4979-8dc5-f45ad02c97cd.mp4

**Reply by rizkhal** [reply]
Currently i use latest version of livewire

**Reply by joshhanley** [reply]
@rizkhal if this is still an issue, can you please confirm the exact patch version of Livewire that your server is using? 

Run `composer show -D` in your server terminal and look for Livewire.

Also any other details, like are you using Octane, etc.

Thanks!

**Reply by rizkhal** [reply]
Hello @joshhanley.
Bellow result from `composer show -D` on my production server

```
barryvdh/laravel-debugbar              v3.6.2     PHP Debugbar integration for Laravel
facade/ignition                        2.14.0     A beautiful error page for Laravel applications.
fakerphp/faker                         v1.16.0    Faker is a PHP library that generates fake data...
fideloper/proxy                        4.4.1      Set trusted proxies for Laravel
fruitcake/laravel-cors                 v2.0.4     Adds CORS (Cross-Origin Resource Sharing) heade...
guzzlehttp/guzzle                      7.3.0      Guzzle is a PHP HTTP client library
laravel-frontend-presets/tall          v3.1.0     TALL preset for Laravel.
laravel-notification-channels/telegram 0.5.1      Telegram Notifications Channel for Laravel
laravel/framework                      v8.62.0    The Laravel Framework.
laravel/sail                           v1.11.0    Docker files for running a basic Laravel applic...
laravel/tinker                         v2.6.2     Powerful REPL for the Laravel framework.
league/flysystem-aws-s3-v3             1.0.29     Flysystem adapter for the AWS S3 SDK v3.x
livewire/livewire                      v2.6.7     A front-end framework for Laravel.
mockery/mockery                        1.4.4      Mockery is a simple yet flexible PHP mock objec...
nunomaduro/collision                   v5.10.0    Cli error handling for console/command-line PHP...
phpunit/phpunit                        9.5.10     The PHP Unit Testing framework.
rappasoft/laravel-livewire-tables      v1.15.0    A dynamic table component for Laravel Livewire
rizkhal/laravel-backup-notification    dev-master A laravel telegram notification
spatie/laravel-backup                  6.16.5     A Laravel package to backup your application
spatie/laravel-fractal                 5.8.1      An easy to use Fractal integration for Laravel ...
spatie/laravel-permission              4.4.1      Permission handling for Laravel 6.0 and up
```

**Reply by joshhanley** [reply]
@rizkhal hmm all looks ok. You're not running Octane/Swoole/Roadrunner on your server are you?

**Reply by rizkhal** [reply]
i just using nginx

bellow the example of my config

```
server {
	listen      80;
       server_name mysite.com;
       root        /var/www/html/mysite.com/public;
       index       index.php;

       charset utf-8;
       gzip on;
	gzip_types text/css application/javascript text/javascript application/x-javascript  image/svg+xml text/plain text/xsd text/xsl text/xml image/x-icon;
        location / {
        	try_files $uri $uri/ /index.php?$query_string;
        }

        location ~ \.php {
                include fastcgi.conf;
                fastcgi_split_path_info ^(.+\.php)(/.+)$;
                fastcgi_pass unix:/run/php-fpm/www.sock;
        }
        location ~ /\.ht {
                deny all;
        }
}
```

**Reply by joshhanley** [reply]
@rizkhal ok, no worries. That all looks ok. Seems strange, it's like it's still running code from before that patch. I don't know why that would be (unless there is another issue with it of course).

**Reply by rizkhal** [reply]
if anyone is experiencing almost the same problem, the solution is in my answer [here](https://github.com/rappasoft/laravel-livewire-tables/discussions/583#discussioncomment-1952233)

**Reply by joshhanley** [reply]
@rizkhal thanks for sharing! 🙂
